ruma-push-gateway-api/src/notify/send_event_notification/v1.rs

```diff
@@ -0,0 +1,176 @@
+//! [POST /_matrix/push/v1/notify](https://matrix.org/docs/spec/push_gateway/r0.1.1#post-matrix-push-v1-notify)
+
+use js_int::UInt;
+use ruma_api::{ruma_api, Outgoing};
+use ruma_client_api::r0::push::PusherData;
+use ruma_events::EventType;
+use ruma_identifiers::{EventId, RoomAliasId, RoomId, UserId};
+use serde::{Deserialize, Serialize};
+use serde_json::Value as JsonValue;
+use std::{collections::BTreeMap, time::SystemTime};
+
+ruma_api! {
+    metadata: {
+        description: "Notify a push gateway about an event or update the number of unread notifications a user has",
+        name: "send_event_notification",
+        method: POST,
+        path: "/_matrix/push/v1/notify",
+        rate_limited: false,
+        requires_authentication: false,
+    }
+
+    #[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+    request: {
+        /// Information about the push notification
+        notification: Notification<'a>,
+    }
```

Comment:
There should be a trailing comma after the last field, and all fields should be public.